docs_src/design/adr/device-service/0021-Profile-Changes.md

```diff
@@ -0,0 +1,126 @@
+
+### Summary of Allowed changes
+
+A detailed look at the allowed changes to device profile and its device resources is listed below.  In general, here is what we are trying to allow/prevent via these rules:
+- Block the DELETE of a device profile when it is associated to any other EdgeX object (device, reading, event) **always**.
+- Allow changes to device profile or device resource fields when the field is inconsequential to the object's use (ex: description) or when it is not yet associated to another EdgeX object.  These changes are listed as `Anytime` or `Situational` accordingly in the table below.
+- Allow the addition of new profiles, device resources, device commands, etc. **always**.  Additions may be accomplished via POST or PATCH depending on the change and API.
+
+See [Proposed Design section](#proposed-design) section below for specific metadata API changes.
+
+While not explicitly covered in this ADR, a device is not allowed when it is associated to an event or reading.
+
+### Profile Changes
+
+The table below outlines the current elements of the EdgeX device profile, and the circumstances allowed for change (accomplished via PUT update API call - whether that is add, update or delete elements where applicable).
+
+| Device Profile Property | Type | Allowed Change | Description/Comments |
+|-------------------------|------|----------------|----------------------|
+| *General Properties* ||||
+| Id | string | Never | This uuid is generated by EdgeX and uniquely identifies the profile. |
+| Name | string | Never | This required property must be unique to the EdgeX instance and cannot be changed.  The profile name can never be empty and must be unique across the entire EdgeX instance. |
+| Manufacturer | string | Anytime | As this property does not affect any functionality other than profile query, this property can be change at any time – to include setting the string to an empty string. |
+| Description | string | Anytime | As this property does not affect any functionality other than profile query, this property can be change at any time – to include setting the string to an empty string. |
+| Model | string | Anytime | As this property does not affect any functionality other than profile query, this property can be change at any time – to include setting the string to an empty string. |
+| Labels | string array | Anytime | As this property does not affect any functionality other than profile query, this property can be change at any time – to include adding, removing or modifying any of the strings to the array of labels. |
+| Device Resources | array | Situational | When the device profile is not associated to any device, reading or event, the device resource list can be modified (i.e., adding or removing the associated device resources).  Once the profile is associated to a device, reading or event, only new resources can be added to the array.  Deleting is prohibited.|
```

Comment:
**suggest**: ...the device resource list can be modified (i.e., adding, modifying, or removing resources).  Once the profile is associated to a device, reading or event, only new resources can be added to the array.  Deleting or modifying existing resources is prohibited.